components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/JWTTokenIssuer.java

```diff
@@ -519,6 +520,32 @@ protected JWTClaimsSet createJWTClaimSet(OAuthAuthzReqMessageContext authAuthzRe
         return jwtClaimsSet;
     }
 
+    /**
+     * Calculates access token expiry time.
+     *
+     * @param accessTokenLifeTimeInMillis accessTokenLifeTimeInMillis
+     * @param curTimeInMillis             currentTimeInMillis
+     * @return expirationTime
+     */
+    private Date calculateAccessTokenExpiryTime(Long accessTokenLifeTimeInMillis, Long curTimeInMillis) {
+
+        Date expirationTime;
+        if (accessTokenLifeTimeInMillis < 0) {
```

Comment:
As per the issue description, I assume accessTokenLifeTimeInMillis is the Token Expiry Time to -1  in the erroneous scenario. Hence why does the variable name says accessTokenLifeTimeInMillis without specifying it as accessTokenExpiryTime